### PR TITLE
CCSDS OPM Writer

### PR DESCRIPTION
**Pull request by HankG**
Adds an OPM writer to take our OrbitParameterMessage objects and write them out to CCSDS formatted OPM messages.  This is based off of 29-ccsds-oem-writer branch which is still in PR since it can take advantage of some of the code in there. This is necessary to complete the OpenOrb OD command line tools.

This should re-target hg-od-hackathon after 29-ccsds-oem-writer branch is merged.

### CODE REVIEW

File: src/main/java/org/b612foundation/adam/opm/OpmWriter.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  public static String toCcsdsOpmString(OrbitParameterMessage opm) {
+    StringBuilder builder = new StringBuilder();
+
+    OdmCommonHeader header = opm.getHeader();
+    builder.append("CCSDS_OPM_VERS = " + opm.getCcsds_opm_vers() + "\n");
+    builder.append("CREATION_DATE = " + header.getCreation_date() + "\n");
+    builder.append("ORIGINATOR = " + header.getOriginator() + "\n");
+
+    OdmCommonMetadata metadata = opm.getMetadata();
```
Comment (HoodedCrow) [inline src/main/java/org/b612foundation/adam/opm/OpmWriter.java]: Paranoia here and everywhere: might want to check for null and maybe throw something like IllegalArgument instead?

File: src/main/java/org/b612foundation/adam/opm/OpmWriter.java [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    builder.append("OBJECT_ID = " + metadata.getObject_id() + "\n");
+    builder.append("CENTER_NAME = " + metadata.getCenter_name() + "\n");
+    builder.append("REF_FRAME = " + metadata.getRef_frame() + "\n");
+    if(metadata.getRef_frame_epoch() != null && !metadata.getRef_frame_epoch().isEmpty()) {
+      builder.append("REF_FRAME_EPOCH = " + metadata.getRef_frame_epoch() + "\n");
+    }
+    builder.append("TIME_SYSTEM = " + metadata.getTime_system() + "\n");
+
+    StateVector stateVector = opm.getState_vector();
+    for(String comment : stateVector.getComments()) {
```
Comment (HoodedCrow) [inline src/main/java/org/b612foundation/adam/opm/OpmWriter.java]: Style nit: space in "for (" and "if (".

File: src/main/java/org/b612foundation/adam/opm/OpmWriter.java (hunk 1 above)
Comment (HankG) [reply src/main/java/org/b612foundation/adam/opm/OpmWriter.java]: I added checks here and in other places to make it more robust and informative.

File: src/main/java/org/b612foundation/adam/opm/OpmWriter.java (hunk 2 above)
Comment (HankG) [reply src/main/java/org/b612foundation/adam/opm/OpmWriter.java]: Fixed in present version.

**Pull request closed without merging**